syntax = "proto3";

service Create_Soldier{
    rpc create_soldiers(stream soldier_details) returns (soldier_output) {}
}

message soldier_details
{
    int64 soldier_number = 1;
    int64 x_pos = 2;
    int64 y_pos = 3;
    int64 speed_capacity = 4;
}

message soldier_output
{
    string msg = 1;
}